filter/filter.go

```diff
@@ -606,6 +610,67 @@ func (c *Client) UpdateBlueprint(ctx context.Context, userAuthToken, serviceAuth
 	return m, eTag, nil
 }
 
+// SubmitFilter function to submit the request to submit a filter for a cantabular dataset.
+// Should POST to /filters/{filterid}/submit in dp-cantabular-filter-flex-api microservice.
+func (c *Client) SubmitFilter(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceToken, ifMatch string, sfr SubmitFilterRequest) (*SubmitFilterResponse, string, error) {
+	b, err := json.Marshal(sfr)
+	if err != nil {
+		return nil, "", err
+	}
+
+	uri := fmt.Sprintf("%s/filters/%s/submit", c.hcCli.URL, sfr.FilterID)
+
+	clientlog.Do(ctx, "updating filter job", service, uri, log.Data{
+		"method": postMethod,
+		"body":   string(b),
+	})
+
+	req, err := http.NewRequest(postMethod, uri, bytes.NewBuffer(b))
+	if err != nil {
+		return nil, "", err
+	}
+
+	if err = headers.SetAuthToken(req, userAuthToken); err != nil {
+		return nil, "", fmt.Errorf("failed to set auth token: %w", err)
```

Comment:
We should be using (github.com/pkg/errors) .Wrap() rather than fmt.Errorf to preserve the stack trace. 
```suggestion
		return nil, "", errors.wrap(err, "failed to set auth token")
```

Reply:
This has been addressed.